homework-submission/week1-project/src/App.js

```diff
@@ -2,34 +2,61 @@ import React, { Component } from 'react';
 import logo from './logo.svg';
 import './App.css';
 import Description from './Components/todoComponents.js'
-
+import TodoForm from './Components/addTodoForm'
+import JSONList from './JSONList'
 
 class App extends Component {
   
   constructor(props){
     super(props);
-    this.state = {list: props.JSONList};
+    this.state = {list: JSONList};
   }
 
   changeState = (id) => {
-    const JSONList = this.state.list;
+    const JSONList = this.state.list.slice();
```

Comment:
Why do call this parameter id while it's NOT?
When you render the Description component you also name id (see line 54).

Because it's now quite confusing to look at JSONList[id] and not to think that the structure of JSONList is like
{ '23242342': { ... }, ... } In which the number is the id and its value is the todo. Which actually is a very nice structure and for the first glance, I thought this's how you've made it!

So, either chaneg the structure, or pass the actual id of each todo(in line 54) and then implement appropriate array methods(e.g. const todoShouldBeEdited = JSONList.filter(todo => todo.id === id) Again I'd name JSONList differently as I mentionsed above.